### PR TITLE
build: add travis deploy

### PR DESCRIPTION
**Pull request by yangchoo**
Adds a travis deploy job to automate deployment to pypi and github
releases

### CODE REVIEW

**Comment by yangchoo** [comment]
its hard to test this specifically, but I've tried playing around with these configuration settings in another repo and it seems to be working fine.

this was done mostly by following best practices listed in the travis ci pages:
- https://docs.travis-ci.com/user/deployment/releases/
- https://docs.travis-ci.com/user/deployment/pypi/

Note that the tokens are encrypted keys with only write access scoped to this particular repository/project
(see https://docs.travis-ci.com/user/encryption-keys/ for more details)

File: .travis.yml [hunk 1]
```diff
@@ -1,8 +1,33 @@
...
+  file: dist/*
+  file_glob: true
+  api_key:
+    secure: 'QYPnx6ABhKhhCCTNEsDCIQ9PrjOXO8ZydrswljVQEjgVDNMSx4FBB586sYeBa8K37b6QOYvGSKM9dZGtKj5+ip1FfgyxICYQWw6LCRT0XdqBB2lXbK9mAErGZbHmx8EQP/xcZ9d+Ta0QpmY1vFbpGWrK+xFCiqKgPHROLQxE2C5bK106bacms4WeRnW/CFaD1qLNtZOYhzehfd2t+tDRN03Mx7aBMsaZLwYLDg4VuvngRsXCtWF7RBwKObcQ6js9HZ+km/R5PE5QSy8FlWy7UKnu/qZ40Ulg7OERx8Q/Kp6ETtAwkW1Fb8BuzbStM/3bPaoQiuBJ2/TBs3mttnUhSGfEJ8ZmlJR7xEpqZ89Il884p+kORAnHwe4JReKvEbSyl/IyG67kOTnfBCoc0KW0qmoLrsqBRo8i35K4i4xQu8H2c0F8g/KXMd6dzKqIjc3flPlAieFl0RVbXNdBe5sD6QkPRXzFGHV23nBsSJkUlVcMcI00xd8x4FlD6cihp1UDsW//LbQArvAXBxw2uJLFZFa5WS2PMU/iENZ8GluJyD/Q277fMN1Jha+fab1G0D8ZQZvj1emZim13zOQ/cREWf9DFjkxBYPpDK5BTQtaUtaHyIhZQS8r/yFJJeDaBr6drNzHB6PYO6ZcufeIqcx7QNVCpdOhRFVYyuqvRZBAOcok='
+  on:
+    repo: strateos/transcriptic
+    tags: true
+    python: '3.6'
+    condition: '$TRAVIS_TAG =~ ^v[0-9]+\.[0-9]+\.[0-9]+$'
+- provider: pypi
```
Comment (dzleidig) [inline .travis.yml:24]: Do we need `skip_existing=true`, as have multiple python environments specified?
https://docs.travis-ci.com/user/deployment/pypi/#upload-artifacts-only-once

Comment (yangchoo) [reply .travis.yml:24]: yup! resolving this

File: .travis.yml [hunk 2]
```diff
@@ -1,8 +1,33 @@
...
 script: tox
+before_deploy:
+- python setup.py sdist
+- python setup.py bdist_wheel
+deploy:
+- provider: releases
+  skip_cleanup: true
+  file: dist/*
+  file_glob: true
+  api_key:
```
Comment (rhysormond) [inline .travis.yml:17]: Hmm I assume this is travis-encrypted?

File: .travis.yml [hunk 3]
```diff
@@ -1,8 +1,33 @@
...
+    secure: 'QYPnx6ABhKhhCCTNEsDCIQ9PrjOXO8ZydrswljVQEjgVDNMSx4FBB586sYeBa8K37b6QOYvGSKM9dZGtKj5+ip1FfgyxICYQWw6LCRT0XdqBB2lXbK9mAErGZbHmx8EQP/xcZ9d+Ta0QpmY1vFbpGWrK+xFCiqKgPHROLQxE2C5bK106bacms4WeRnW/CFaD1qLNtZOYhzehfd2t+tDRN03Mx7aBMsaZLwYLDg4VuvngRsXCtWF7RBwKObcQ6js9HZ+km/R5PE5QSy8FlWy7UKnu/qZ40Ulg7OERx8Q/Kp6ETtAwkW1Fb8BuzbStM/3bPaoQiuBJ2/TBs3mttnUhSGfEJ8ZmlJR7xEpqZ89Il884p+kORAnHwe4JReKvEbSyl/IyG67kOTnfBCoc0KW0qmoLrsqBRo8i35K4i4xQu8H2c0F8g/KXMd6dzKqIjc3flPlAieFl0RVbXNdBe5sD6QkPRXzFGHV23nBsSJkUlVcMcI00xd8x4FlD6cihp1UDsW//LbQArvAXBxw2uJLFZFa5WS2PMU/iENZ8GluJyD/Q277fMN1Jha+fab1G0D8ZQZvj1emZim13zOQ/cREWf9DFjkxBYPpDK5BTQtaUtaHyIhZQS8r/yFJJeDaBr6drNzHB6PYO6ZcufeIqcx7QNVCpdOhRFVYyuqvRZBAOcok='
+  on:
+    repo: strateos/transcriptic
+    tags: true
+    python: '3.6'
+    condition: '$TRAVIS_TAG =~ ^v[0-9]+\.[0-9]+\.[0-9]+$'
```
Comment (rhysormond) [inline .travis.yml:23]: Not sure what flavor of regex Travis uses but can these just be `\d` instead of `[0-9]`?

Comment (dauglyon) [reply .travis.yml:23]: `\d` will sometimes matches all unicode digits (e.g. 𑄷, 𑄸, 𑄿). I tend towards [0-9] for safety.

File: .travis.yml (hunk 2 above)
Comment (yangchoo) [reply .travis.yml:17]: yup these are both encrypted keys that are write-scoped

File: .travis.yml (hunk 3 above)
Comment (rhysormond) [reply .travis.yml:23]: wow. wait. what?

**Pull request merged**